tests/test_types.py

```diff
@@ -4960,7 +4960,7 @@ class Model(BaseModel):
 
 def test_smart_union_compounded_types_edge_case():
     class Model(BaseModel):
-        x: Union[List[str], List[int]]
+        x: 'List[str] | List[int]'
```

Comment:
You didn't change _all_ unions did you? We should probably make sure we keep at least some instances of `Union[X, Y]` in tests to make sure we can catch any issues that might occur for that but not for `X | Y`. (Even if that shouldn't be an issue today, it might end up being an issue with future code changes.)

Reply:
No, there's still loads of normal Unions left.